### PR TITLE
fms2io: domain write fix

### LINKED ISSUE #447: FMS 2020.03-alpha1 cause crashes in diag_manager_end with NetCDF: Numeric conversion not representable
**Describe the bug**
Restart regression tests of two different models crash in diag_manager_end in the second leg of experiemnt with the following traceback: 

```
FATAL from PE  1728: NetCDF: Numeric conversion not representable

Image              PC                Routine            Line        Source
fms_esm4.1_sis2_c  00000000053606F6  Unknown               Unknown  Unknown
fms_esm4.1_sis2_c  000000000337C913  mpp_mod_mp_mpp_er          68  mpp_util_mpi.inc
fms_esm4.1_sis2_c  00000000031145D0  fms_io_utils_mod_         164  fms_io_utils.F90
fms_esm4.1_sis2_c  00000000030B872B  netcdf_io_mod_mp_         284  netcdf_io.F90
fms_esm4.1_sis2_c  00000000034A69CA  fms_netcdf_domain         834  domain_write.inc
fms_esm4.1_sis2_c  000000000348619F  diag_output_mod_m        1270  diag_output.F90
fms_esm4.1_sis2_c  0000000003195648  diag_util_mod_mp_        2538  diag_util.F90
fms_esm4.1_sis2_c  0000000003461CF0  diag_manager_mod_        3542  diag_manager.F90
fms_esm4.1_sis2_c  000000000345F402  diag_manager_mod_        3760  diag_manager.F90
fms_esm4.1_sis2_c  0000000000416906  coupler_main_IP_c        1995  coupler_main.F90
fms_esm4.1_sis2_c  000000000040CD9D  MAIN__                   1151  coupler_main.F90
```

**To Reproduce**
```
git clone -b RTS https://gitlab.gfdl.noaa.gov/mdt/xml mdt_xml_FMS2020.03
cd mdt_xml_FMS2020.03
fremake -x OM4_COBALT_postxanadu.xml -p ncrc4.intel18 -t prod MOM6_SIS2_compile
frerun -x OM4_COBALT_postxanadu.xml -p ncrc4.intel18 -t prod OM4p5_CORE2_IAF_COBALT_abio_csf_mle200 -r rts --no-transfer -u -s
```

**Expected behavior**
RTS should not crash!

**System Environment**
gaea and orion

**Additional context**
On gaea I see this only if the test has a restart in the middle, but on orion it happens on 1 segment  runs too. Also, I see this only in tests using a ocean mask_table, so this might be @ngs333 old friend issue #409  back in a new suit. 

ESM4 crashes on gaea:
```
/lustre/f2/scratch/Niki.Zadeh/FMS2020.03-alpha1min_mom6_ESM4_v1.0.3min_1/ESM4_piControl_D/ncrc4.intel18-prod-openmp/stdout/run/ESM4_piControl_D_2x0m15d_1728x4a_1437x1o.o268648259
```
OM4 crashes on gaea:
```
/lustre/f2/scratch/Niki.Zadeh/FMS2020.03-alpha1_mom6_ESM4_v1.0.3_min_0604_1/OM4p5_CORE2_IAF_COBALT_abio_csf_mle200/ncrc4.intel18-prod/stdout/run/OM4p5_CORE2_IAF_COBALT_abio_csf_mle200_2x0m15d_1057x1o.o268647635
```
OM4 crashes on orion:
```
OK
/work/noaa/gfdlscr/nzadeh/FMS2020.03-alpha1_mom6_ESM4_v1.0.3_min/OM4p5_CORE2_IAF_COBALT_abio_csf_mle200/orion.intel18-prod/stdout/run/OM4p5_CORE2_IAF_COBALT_abio_csf_mle200_1x0m2d_442x1o.o73362 

Crash
/work/noaa/gfdlscr/nzadeh/FMS2020.03-alpha1_mom6_ESM4_v1.0.3_min/OM4p5_CORE2_IAF_COBALT_abio_csf_mle200/orion.intel18-prod/stdout/run/OM4p5_CORE2_IAF_COBALT_abio_csf_mle200_1x0m2d_442x1o6.o73533
```

### PR DESCRIPTION
**Pull request by uramirez8707**
**Description**
This PR:
1. Initializes the global buffer to the fill_value that is in the file 
2. Uses the global io domain to allocate the global buffer
3. Adds a unit test that writes a domain decomposed file with a mask

Fixes #447 

**How Has This Been Tested?**
Ran OM4p5_CORE2_IAF_COBALT_abio_csf_mle200 on GAEA with intel18 and prod: 

git clone -b RTS https://gitlab.gfdl.noaa.gov/mdt/xml mdt_xml_FMS2020.03
cd mdt_xml_FMS2020.03
fremake -x OM4_COBALT_postxanadu.xml -p ncrc4.intel18 -t prod MOM6_SIS2_compile
-------- Need to pull in my FMS changes or modify the xml to checkout out my branch
frerun -x OM4_COBALT_postxanadu.xml -p ncrc4.intel18 -t prod OM4p5_CORE2_IAF_COBALT_abio_csf_mle200 -r rts --no-transfer -u -s

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by uramirez8707** [comment]
fyi @ngs333 

Please take a look @menzel-gfdl, @GFDL-Eric

**Comment by rem1776** [comment]
tested this on the skylake with intel/2020 and found no issues

**Comment by ngs333** [comment]
Uriel,
I will test on Gaea, hopefully today!

**Comment by ngs333** [comment]
I've been able to test quite a bit and the test results are that for a given PE count, the xanadu history files are reproduced by the master+fix history files; and likewise for the restart files. This is for both tests that use the masking tables and tests that don't use them.

**Comment by edwardhartnett** [comment]
Can a test be added to verify the fix?

File: test_fms/fms2_io/test_io_with_mask.F90 [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+!! data when the domain contains a mask table. For the points that are
+!! masked out, no data should be writen.
+
+use   mpp_domains_mod
+use   mpp_mod
+use   fms2_io_mod
+use   fms_mod, only: fms_init, fms_end
+use   fms_io_mod, only: parse_mask_table
+use   netcdf
+use,  intrinsic :: iso_fortran_env, only : real64
```
Comment (thomas-robinson) [inline test_fms/fms2_io/test_io_with_mask.F90:32]: You have an inconsistent use of `only`

**Pull request merged**